code/modules/mob/mob.dm

```diff
@@ -1331,6 +1331,29 @@ var/list/slot_equipment_priority = list( \
 /mob/proc/is_literate()
 	return FALSE
 
+/mob/proc/faction_check_mob(mob/target, exact_match)
+	if(exact_match) //if we need an exact match, we need to do some bullfuckery.
+		var/list/faction_src = faction.Copy()
+		var/list/faction_target = target.faction.Copy()
+		if(!("\ref[src]" in faction_target)) //if they don't have our ref faction, remove it from our factions list.
+			faction_src -= "\ref[src]" //if we don't do this, we'll never have an exact match.
+		if(!("\ref[target]" in faction_src))
+			faction_target -= "\ref[target]" //same thing here.
+		return faction_check(faction_src, faction_target, TRUE)
+	return faction_check(faction, target.faction, FALSE)
+
+/proc/faction_check(list/faction_A, list/faction_B, exact_match)
+	var/list/match_list
+	if(exact_match)
+		match_list = faction_A & faction_B //only items in both lists
+		var/length = LAZYLEN(match_list)
+		if(length)
+			return (length == LAZYLEN(faction_A)) //if they're not the same len(gth) or we don't have a len, then this isn't an exact match.
```

Comment:
What would happen if `faction_A` was a subset of the items in `faction_B`?
`match_list` would be the same as `faction_A`, but `faction_B` would contain items not found in `faction_A` - but the check would still pass.
If `faction_B` and `faction_A` in this scenario were swapped, then the check would fail, since `LAZYLEN(faction_B) > LAZYLEN(match_list)`

Reply:
I think you're right. But I think we'll just have to take care that won't happen - it hasn't caused any issues on /tg/ so far.